File: backend/src/main/java/com/finsight/datahub/service/UploadService.java
package com.finsight.datahub.service;

import com.finsight.datahub.dto.response.UploadHistoryDto;
import com.finsight.datahub.dto.response.UploadResponseDto;
import com.finsight.datahub.entity.AssetType;
import com.finsight.datahub.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.io.InputStream;

public interface UploadService {

    UploadResponseDto uploadCsv(MultipartFile file, AssetType assetType, User user);

    UploadResponseDto uploadFileStream(InputStream inputStream, String originalFilename, long fileSizeBytes, AssetType assetType, User user);

    Page<UploadHistoryDto> getUploadHistory(Pageable pageable);

    UploadHistoryDto getUploadDetails(Long id);
}

File: frontend/src/services/uploadService.js
import api from './api';

export const uploadService = {
  uploadCsv: async (file, assetType) => {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('assetType', assetType);

    const response = await api.post('/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },

  getUploadHistory: async () => {
    const response = await api.get('/upload/history');
    return response.data;
  },

  getUploadDetails: async (id) => {
    const response = await api.get(`/upload/${id}`);
    return response.data;
  },
};
